File: Source/Core/Utility/NSCharacterSet+Utility.m
//
//  NSCharacterSet+Utility.m
//  Eunomia
//
//  Created by Ian on 10/20/14.
//  Copyright (c) 2014 Eunomia. All rights reserved.
//

#import "NSCharacterSet+Utility.h"

@implementation NSCharacterSet (Eunomia_Utility)

- (NSString *)stringWithCharactersInSet
{
    NSMutableString *result = [NSMutableString string];
    
    for (unichar test = 0; test < UCHAR_MAX; test++)
    {
        if ( [self characterIsMember:test] )
        {
            [result appendFormat:@"%c", test];
        }
    }
    
    return result;
}

@end

File: Source/Core/UtilityExtensions/NSCharacterSet+Utility.swift
//
//  NSCharacterSet+Utility.swift
//  Eunomia
//
//  Created by Ian Grossberg on 11/26/15.
//  Copyright © 2015 Adorkable. All rights reserved.
//

import Foundation

extension CharacterSet {
    
    // TODO: ascending and descending
    public var charactersInSetAscending : String {
        var result = String()

        for test in unichar.min...unichar.max {
            if let unicodeValue = UnicodeScalar(test) {
                if self.contains(unicodeValue) {

                    var found = test
                    withUnsafePointer(to: &found, { (testPointer) -> Void in
                        let append = NSString(characters: testPointer, length: 1)
                        result = result + (append as String)
                    })
                }
            }
        }
        
        return result
    }
}
